### DISCUSSION TITLE
Images tags and versions

### DISCUSSION THREAD

**Discussion by nicoschi** [docksal/docksal#1589] [Q&A] [answered]
Hi there, just a question about Docksal images tags system. 

Is it right to consider these tags are pointing to the same image? 
Examples:
php8.0-[latest_major].[latest_minor] === php8.0 (in docksal/cli)
[latest_major].[latest_minor]-php7.2 === php7.2 (in docksal/cli)
[latest_major].[latest_minor]-mysql-5.6 === mysql-5.6 (in docksal/db)
[latest_major].[latest_minor]-apache2.4 === apache2.4 (in docksal/web)
[latest_major].[latest_minor]-6.2 === 6.2 (in docksal/varnish)
... and so on?

Thanks

**Comment by lmakarov** [comment, marked as answer]
The general tagging pattern for all Docksal images is the following:

```
<image-repo>:<software-version>[-<image-stability-tag>][-<flavor>]
```

See https://docs.docksal.io/stack/images-versions#understanding-versions

The following image tags (using docksal/cli as an example) would point to the same image:

```
docksal/cli:php8.0-3.0
docksal/cli:php8.0-3
docksal/cli:php8.0
```

We do not tag hotfix releases, so a minor version tag (e.g., `docksal/cli:php8.0-3.0`) will move if there was a hotfix release (e.g., v3.0.1).

Note: with `docksal/cli` v3 we aligned the image tagging pattern with other Docksal images.
docksal/cli v2 images used to follow a slightly different pattern (v2.13 is the latest version in the v2 series):

```
docksal/cli:2.13-php7.4
docksal/cli:2-php7.4
docksal/cli:php7.4 - this tag points to a v3 image at this point
```

**Reply by nicoschi** [reply]
Thank you for your answer. I missed that section of the documentation which is already exhaustive itself.
